streaming/index.js

```diff
@@ -677,7 +756,12 @@ const startServer = async () => {
         eventHandlers.onKill();
       } else if (event === 'filters_changed') {
         req.log.debug(`Invalidating filters cache for ${req.accountId}`);
-        req.cachedFilters = null;
+
+        fetchFilters(req.accountId).then((filters) => {
+          req.cachedFilters = filters;
```

Comment:
This does mean that the previous filters live for the time it takes to round trip to the database to recalculate the filters, but I think that's okay. I think it's better to apply the previous filters whilst we refresh them in the background.

Reply:
Arguably we could make this more efficient if the `filters_changed` event carried with it which filter was changed, though that requires changes on the ruby side, and only really improves performance if you've a large number of filters, and even then they're not likely to change that often.